zenoh/tests/shm.rs

```diff
@@ -124,8 +145,18 @@ async fn test_session_pubsub(peer01: &Session, peer02: &Session, reliability: Re
         let _sub = ztimeout!(peer01
             .declare_subscriber(&key_expr)
             .callback(move |sample| {
-                assert_eq!(sample.payload().len(), size);
-                let _ = sample.payload().as_shm().unwrap();
+                let expected_size = if resize_buffer != ResizeBuffer::NoResize {
```

Comment:
What if `resize_buffer` == `ResizeBuffer::Relayout`? We could document the cases covered in this test above the enum definition. Like 

1. NoResize: Preserve the original buffer size.
2. Resize: Halve the buffer size during the test.
3. Relayout:  Halve the buffer size and change the layout during the test.